File: src/main/java/com/sonet/storage/model/item/Item.java
package com.sonet.storage.model.item;

import com.sonet.storage.model.producer.Producer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;

@Entity
@Table(name = "item")
@NoArgsConstructor
@ToString
@Getter
@Setter
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String code;

    private String size;

    private Integer pack;

    private Double price;

    private String description;

    @ManyToOne
    private Producer producer;

    public Item(String code, String size, Integer pack, Double price, String description, Producer producer) {
        this.code = code;
        this.size = size;
        this.pack = pack;
        this.price = price;
        this.description = description;
        this.producer = producer;
    }
}

File: src/main/webapp/storage/src/app/dto/Item.ts
import { Producer } from "./Producer";

export class Item {
    private readonly _id: number;
    private readonly _code: string;
    private readonly _size: string;
    private readonly _pack: number;
    private readonly _price: string;
    private readonly _description: string;
    private readonly _producer: Producer;


  constructor(id: number, code: string, size: string, pack: number, price: string, description: string, producer: Producer) {
    this._id = id;
    this._code = code;
    this._size = size;
    this._pack = pack;
    this._price = price;
    this._description = description;
    this._producer = producer;
  }


  get id(): number {
    return this._id;
  }

  get code(): string {
    return this._code;
  }

  get size(): string {
    return this._size;
  }

  get pack(): number {
    return this._pack;
  }

  get price(): string {
    return this._price;
  }

  get description(): string {
    return this._description;
  }

  get producer(): Producer {
    return this._producer;
  }
}
